client/app/components/queries/ScheduleDialog.jsx

```diff
@@ -139,6 +139,29 @@ class ScheduleDialog extends React.Component {
     this.setScheduleUntil(null, date);
   };
 
+  getTimeSelector = () => {
```

Comment:
If not for the need to define PropTypes, I think this could be better served by a component than a function.

TypeScript can't come fast enough 😩 I wonder if we should just use an eslint comment to disable the PropTypes check for "inner" components? (it's not like we're checking them in this case anyway)

(this is not a blocker)

Reply:
Done 4156776926ff6510cbed00328d3d6c4f8bfa06ea